pkg/client/fake/client.go

```diff
@@ -319,6 +321,13 @@ func (t versionedTracker) Update(gvr schema.GroupVersionResource, obj runtime.Ob
 	}
 	intResourceVersion++
 	accessor.SetResourceVersion(strconv.FormatUint(intResourceVersion, 10))
+
+	// The behavior of the generation field is inconsistent across built-in k8s resources, but it's
+	// consistent for CRs. CRs have their generation increased on non-metadata changes only.
+	if isCustomResource(gvk) && didNonMetadataFieldChange(oldObject, obj) {
+		accessor.SetGeneration(oldAccessor.GetGeneration() + 1)
```

Comment:
If we are going to enable `Generation` in fakeClient, you should also set it to `1` when object created.
See https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go#L118

Reply:
Done